### PR TITLE
Generate documentation for copyWith and toJson

### LINKED ISSUE #1068: Documenting the generated methods copyWith and toJson
I'm finishing my first flutter plugin where I am using freezed for some data classes. My problem is just that pub.dev marks that I did't document the copyWith and toJson methods. I would like to do that without modifying the generated files.

I would like to see that this part is auto generated or that you give me an option to set that e.g. in a annotation.

### PR DESCRIPTION
**Pull request by rekire**
Resolves #1068

I tried to minimize the changes. This PR also corrects the position of the `@override` for some cases where the override was above the documentation.

### CODE REVIEW

**Comment by rekire** [comment]
@rrousselGit I noticed that you merged a dependabot PR, so you are working today. Is there anything in my PR I should change?

File: packages/freezed/lib/src/templates/copy_with.dart [hunk 1]
```diff
@@ -61,17 +61,19 @@ ${_abstractDeepCopyMethods().join()}
   String get abstractCopyWithGetter {
     if (cloneableProperties.isEmpty) return '';
 
-    return _maybeOverride(
-      '''
-@JsonKey(ignore: true)
+    return '''
```
Comment (rrousselGit) [inline packages/freezed/lib/src/templates/copy_with.dart:64]: Sounds like _maybeOverride should have an optional `doc` parameter. Otherwise LGTM

**Review by rrousselGit — CHANGES_REQUESTED** [review]
There appear to be a few missing places with regards to copyWith documentation.

We need to document the `call` method of the object returned by `get copyWith`.
This will provide doc when writing `copyWith(...)`.
Currently docs appear with `.copyWith`, but not `.copyWiht(...)`

We also will want doc for deep copy, such as `value.copyWith.field(...)`

**Comment by rrousselGit** [comment]
Do you still wish to work on this?

**Comment by rekire** [comment]
To be honest I totally forgot about this PR. I need to check my PR to find out what changes are still required. Might be someone else would be quicker, but I want to finish what I have started. Thank you for the reminder.

**Comment by rekire** [comment]
@rrousselGit did I catch all the missing parts? The last time I checked the generated code everything was documented. I am not sure why I missed those.

**Comment by rrousselGit** [comment]
It sounds good. Do you mind refactoring the doc into a getter on `data`? So that we don't have multiple duplicates of the same doc in the source.

**Comment by rekire** [comment]
@rrousselGit done

File: packages/freezed/lib/src/templates/copy_with.dart [hunk 2]
```diff
@@ -23,6 +23,11 @@ class CopyWith {
     return '\$${name}CopyWith';
   }
 
+  static String _generateDataDocs(String name, {String indent = ''}) {
+    return '$indent/// Create a copy of $name\n$indent'
+        '/// with the given fields replaced by the non-null parameter values.';
+  }
```
Comment (rrousselGit) [inline packages/freezed/lib/src/templates/copy_with.dart:29]:
```suggestion
  String _copyWithDocs(String name) {
    return '''
/// Create a copy of $name
/// with the given fields replaced by the non-null parameter values.''';
  }
```

File: packages/freezed/lib/src/templates/copy_with.dart [hunk 3]
```diff
@@ -63,8 +69,7 @@ ${_abstractDeepCopyMethods().join()}
 
     return _maybeOverride(
       doc: '''
-/// Create a copy of ${data.name}
-/// with the given fields replaced by the non-null parameter values.
+${_generateDataDocs(data.name)}
```
Comment (rrousselGit) [inline packages/freezed/lib/src/templates/copy_with.dart:72]:
```suggestion
${_copyWithDocs(data.name)}
```

File: packages/freezed/lib/src/templates/copy_with.dart [hunk 4]
```diff
@@ -76,8 +81,7 @@ $_abstractClassName${genericsParameter.append('$clonedClassName$genericsParamete
   String get concreteCopyWithGetter {
     if (cloneableProperties.isEmpty) return '';
     return '''
-/// Create a copy of ${data.name}
-/// with the given fields replaced by the non-null parameter values.
+${_generateDataDocs(data.name)}
```
Comment (rrousselGit) [inline packages/freezed/lib/src/templates/copy_with.dart:84]:
```suggestion
${_copyWithDocs(data.name)}
```

File: packages/freezed/lib/src/templates/copy_with.dart [hunk 5]
```diff
@@ -133,8 +137,7 @@ class $_implClassName${genericsDefinition.append('\$Res').append('\$Val extends
   // ignore: unused_field
   final \$Res Function(\$Val) _then;
 
-  /// Create a copy of ${data.name}
-  /// with the given fields replaced by the non-null parameter values.
+${_generateDataDocs(data.name, indent: '  ')}
```
Comment (rrousselGit) [inline packages/freezed/lib/src/templates/copy_with.dart:140]:
```suggestion
${_copyWithDocs(data.name)}
```

File: packages/freezed/lib/src/templates/copy_with.dart [hunk 6]
```diff
@@ -151,8 +154,7 @@ class $_implClassName${genericsDefinition.append('\$Res')} extends ${parent!._im
       : super(_value, _then);
 
 
-  /// Create a copy of ${data.name}
-  /// with the given fields replaced by the non-null parameter values.
+${_generateDataDocs(data.name, indent: '  ')}
```
Comment (rrousselGit) [inline packages/freezed/lib/src/templates/copy_with.dart:157]:
```suggestion
${_copyWithDocs(data.name)}
```

File: packages/freezed/lib/src/templates/copy_with.dart [hunk 7]
```diff
@@ -353,8 +355,7 @@ $constructorParameters
       final cast = isConcrete ? '' : 'as \$Val';
 
       yield '''
-/// Create a copy of ${data.name}
-/// with the given fields replaced by the non-null parameter values.
+${_generateDataDocs(data.name)}
```
Comment (rrousselGit) [inline packages/freezed/lib/src/templates/copy_with.dart:358]:
```suggestion
${_copyWithDocs(data.name)}
```

**Review by rrousselGit — APPROVED** [review]

**Comment by rekire** [comment]
This looks like a bug of my PR or by a change made after this PR. So or so please create a new issue for that. This PR was merged in summer last year.

**Pull request merged**